homeworks/rostyslav-reshmed/homework-6/script-2.js

```diff
@@ -0,0 +1,19 @@
+const obj = { key: { key2: { key: 1 }, value: 3 } };
+
+function sumNumber(obj) {
+    let sum = 0;
+    for (const [key, value] of Object.entries(obj)) {
+        if (value instanceof Object) {
+            sum = sumNumber(value);
+            sum += +key.replace(/[^+\d]/g, '');
+        } else if (Object(value) instanceof Number) {
```

Comment:
```suggestion
        } else if (!isNaN(value)) {
```
я бы так заюзал, не уверен что твой вариант не имеет подводных камней.

Вот нашел один из них `Object(NaN) instanceof Number`